modules/apps/portal-workflow/portal-workflow-kaleo-runtime-impl/src/main/java/com/liferay/portal/workflow/kaleo/runtime/internal/assignment/MultiLanguageKaleoTaskAssignmentSelector.java

```diff
@@ -82,33 +82,38 @@ public Collection<KaleoTaskAssignment> getKaleoTaskAssignments(
 		return kaleoTaskAssignments;
 	}
 
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(scripting.language=*)"
-	)
-	protected void addKaleoTaskAssignmentSelector(
-			ScriptingAssigneeSelector scriptingAssigneeSelector,
-			Map<String, Object> properties)
-		throws KaleoDefinitionValidationException {
-
-		_scriptingassigneeSelectors.put(
-			_getKaleoTaskAssignmentSelectKey(
-				GetterUtil.getString(properties.get("scripting.language")),
-				ClassUtil.getClassName(scriptingAssigneeSelector)),
-			scriptingAssigneeSelector);
+	@Activate
+	protected void activate(BundleContext bundleContext) {
+		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
+			bundleContext, ScriptingAssigneeSelector.class,
+			"(scripting.language=*)",
+			(serviceReference, emitter) -> {
+				String className = ClassUtil.getClassName(
```

Comment:
@W2jx Is it possible to follow what Tina changed in `com.liferay.portal.workflow.kaleo.runtime.internal.action.executor.ActionExecutorManagerImpl` to simplify the key related logic?